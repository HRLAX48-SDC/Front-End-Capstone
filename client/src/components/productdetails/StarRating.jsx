import React, {useState, useEffect} from 'react';
import Nav from 'react-bootstrap/Nav';
import { Rating } from 'react-simple-star-rating';

function StarRating(props) {

  const [avgRating, setAvgRating] = useState(null);
  const [review, setReview] = useState(0);

  useEffect(() => {
    average(props.ratings);
  }, [props.ratings])

  const handleReviewClick = () => {
    let offset = 100;
    window.scrollTo({
        behavior: "smooth",
        top:
        document.getElementById("rating-reviews-main").getBoundingClientRect().top -
        document.body.getBoundingClientRect().top -
        offset
    });
  }

  const average = (ratingObject) => {
    if (Object.keys(ratingObject).length !== 0) {
      var sum = 0;
      var count = 0;
      for (var key in ratingObject) {
        count += Number(ratingObject[key]);
        sum += Number(key) * Number(ratingObject[key]);
      }
      setAvgRating(parseFloat(sum/count).toFixed(1));
      setReview(count);
    }
  }

  return (
    <div>
      {review ?
        <div>
          <Rating
            readonly={true}
            ratingValue={avgRating * 20}
            initialValue={avgRating * 20}
            allowHalfIcon={true}
            size={28}
          />
          <span>{avgRating}</span>
          <span className='p_rating'><a onClick={handleReviewClick}>Read All {review} Ratings & Reviews</a></span>
        </div>
      :
        null
      }
    </div>
  );
}

export default StarRating;