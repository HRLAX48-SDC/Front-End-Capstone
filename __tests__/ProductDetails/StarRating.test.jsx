import React from 'react';
import renderer from 'react-test-renderer';
import StarRating from '../../client/src/components/productdetails/StarRating.jsx';

test('renders StarRating component correctly', () => {
  const starRating = renderer.create(<StarRating
    ratings={{1: '23', 2: '7', 3: '49', 4: '36', 5: '45'}}
  />).toJSON();
  expect(starRating).toMatchSnapshot();
});

test('renders StarRating component correctly', () => {
  const starRating = renderer.create(<StarRating
    ratings={{1: '23', 2: '7', 3: '49', 4: '36', 5: '45'}}
  />).root;

  const rating = starRating.findAllByProps({
    ratingValue: 70
  });
  expect(rating).toBeDefined();
});