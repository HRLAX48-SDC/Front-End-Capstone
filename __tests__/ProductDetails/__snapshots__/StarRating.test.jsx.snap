// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders StarRating component correctly 1`] = `<div />`;
